clang/include/clang/Basic/Attr.td

```diff
@@ -407,7 +407,8 @@ def MicrosoftExt : LangOpt<"MicrosoftExt">;
 def Borland : LangOpt<"Borland">;
 def CUDA : LangOpt<"CUDA">;
 def HIP : LangOpt<"HIP">;
-def SYCL : LangOpt<"SYCLIsDevice">;
+def SYCLHost : LangOpt<"SYCLIsHost">;
```

Comment:
On the other hand SYCL is a trademark from Khronos, so if it can stay SYCL it is better from this perspective.
Otherwise people get confused and create more effort like https://github.com/llvm/llvm-project/pull/113060 :smiley:

Reply:
> On the other hand SYCL is a trademark from Khronos, so if it can stay SYCL it is better from this perspective. Otherwise people get confused and create more effort like #113060 😃

I retract that point anyway, I cannot find where I got that from, it doesn't seem to be in our coding standard, though I thought it was!  So I guess that was just an N-AI (not-artificial intelligence... or Artificial Not-intelligence?  you choose...) hallucination.

Reply:
With regard to the `Is`, I was just preserving the style that was already there. The corresponding command line options are named `-fsycl-is-device` and `-fsycl-is-host`, so perhaps it is worth maintaining that symmetry. I'm ambivalent.

Reply:
Those are a weird choice too IMO.  I think we inherited that without much review from the downstream, but I kind of want to push back.  The `-is` doesn't really buy us anything here.  I'd accept the flag as an alias, though I'm not sure how much value there is as we aren't really functional in the upstream, so our users are limited.

Reply:
I've never loved those names either. I don't think it makes sense to change them as part of this PR though. https://github.com/llvm/llvm-project/pull/107493 is introducing new `-fsycl-device-only` and `-fsycl-host-only` options as aliases of `--offload-device-only` and `--offload-host-only` (also aliased by `--cuda-device-only` and `--cuda-host-only`). It might make sense to argue for a change in that PR. Perhaps there is an opportunity to remove the `-fsycl-is-device` and `-fsycl-is-host` options in favor of a common offload indicator of host vs device compilation.

For the purposes of this PR, I think it makes sense to stick with the status quo pending a proposal for a better, perhaps unified, way of handling host vs device compilation.

Reply:
> I've never loved those names either. I don't think it makes sense to change them as part of this PR though. #107493 is introducing new `-fsycl-device-only` and `-fsycl-host-only` options as aliases of `--offload-device-only` and `--offload-host-only` (also aliased by `--cuda-device-only` and `--cuda-host-only`). It might make sense to argue for a change in that PR. Perhaps there is an opportunity to remove the `-fsycl-is-device` and `-fsycl-is-host` options in favor of a common offload indicator of host vs device compilation.
> 
> For the purposes of this PR, I think it makes sense to stick with the status quo pending a proposal for a better, perhaps unified, way of handling host vs device compilation.

Please lobby this there.  I'm OK with leaving it in the flag name here, but I still would like the lang-opt changed here.

Reply:
The `SYCLIsDevice` and `SYCLIsHost` members of `LangOpt` are pre-existing and referenced from numerous source files. I'm not going to change them as part of this PR. You are welcome to submit a PR to change them if you like. Otherwise, we can investigate changing them as part of a future offload unification effort.

Reply:
To me, this is a "longer we wait, the harder this is". So no time like the present!

Reply:
IMO, where we eventually want to end up is with `IsDevice` and `IsHost` which composes with all the offloading languages (I cannot imagine a situation where you're compiling for the device in one offloading language while compiling for the host in another offloading language in the same compilation pass). e.g., `getLangOpts().SYCL && getLangOpts().IsDevice` tells you that you're doing a device compilation for SYCL.

Reply:
Aaron chatted with me off-line, and I think THAT is the eventual place we want to be.  So doing this work now, while beneficial, would eventually become wasted work.  So I'm resolving this.

Reply:
Thanks, Erich. The model Aaron described matches what I had in mind as part of some future offload unification effort, so it sounds like we are all aligned here.